scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -748,7 +761,116 @@ private void deleteActiveSchedules(String appId) {
+					&& policySchedulesMap.get(appIdInPolicy).getSchedules().hasSchedules()) {
+				if (appIdAndGuidMap.get(appIdInPolicy) == null) {
+					toCreateScheduleList.add(policySchedulesMap.get(appIdInPolicy));
+					createCount++;
+					continue;
+				} else if (!scheduleAppIdGuidMap.get(appIdInPolicy).equals(appIdAndGuidMap.get(appIdInPolicy))) {
+					toCreateScheduleList.add(policySchedulesMap.get(appIdInPolicy));
+					toDeletedAppIds.add(appIdInPolicy);
+					updateCount++;
+					continue;
+				}
+			}
+		}
+		
+		Set<String> appIdInScheduleSet = new HashSet<String>();
+		Set<String> appIdInPolicySet = policySchedulesMap.keySet();
+		appIdInScheduleSet.addAll(appIdAndGuidMap.keySet());
+		for(String appId : appIdInScheduleSet){
+			if(!appIdInPolicySet.contains(appId)){
+				toDeletedAppIds.add(appId);
+				deleteCount++;
+			}
+		}
+		for(String appId : toDeletedAppIds){
+			this.deleteSchedules(appId);
+		}
+		
+		for(ApplicationSchedules schedule : toCreateScheduleList){
+			this.createSchedules(schedule.getSchedules());
+		}		
+		
+		return new SynchronizeResult(createCount,updateCount,deleteCount);
+		
+	}
+	
+	private ApplicationSchedules parseSchedulesFromPolicy(PolicyJsonEntity policyJsonEntity){
+		ObjectMapper mapper = new ObjectMapper();
+		String policyJson = policyJsonEntity.getPolicyJson();
+		ApplicationSchedules applicationSchedules = null;
+		mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+		try {
+			applicationSchedules = mapper.readValue(policyJson, ApplicationSchedules.class);
+			if(applicationSchedules!=null && applicationSchedules.getSchedules()!=null&&applicationSchedules.getSchedules().hasSchedules()){
+				Schedules schedules = applicationSchedules.getSchedules();
+				List<RecurringScheduleEntity> recurringSchedules = schedules.getRecurringSchedule();
+				List<SpecificDateScheduleEntity> specificDateSchedules = schedules.getSpecificDate();
+				if(recurringSchedules != null){
+					for(RecurringScheduleEntity recurring : recurringSchedules){
+						recurring.setAppId(policyJsonEntity.getAppId());
+						recurring.setTimeZone(schedules.getTimeZone());
+						recurring.setDefaultInstanceMinCount(applicationSchedules.getInstanceMinCount());
+						recurring.setDefaultInstanceMaxCount(applicationSchedules.getInstanceMaxCount());
+						recurring.setGuid(policyJsonEntity.getGuid());
+					}
+				}
+				if(specificDateSchedules != null){
+					for(SpecificDateScheduleEntity specificDate : specificDateSchedules){
+						specificDate.setAppId(policyJsonEntity.getAppId());
+						specificDate.setTimeZone(schedules.getTimeZone());
+						specificDate.setDefaultInstanceMinCount(applicationSchedules.getInstanceMinCount());
+						specificDate.setDefaultInstanceMaxCount(applicationSchedules.getInstanceMaxCount());
+						specificDate.setGuid(policyJsonEntity.getGuid());
+					}
+				}
+				 
+			}
+			
+		} catch (IOException e) {
+			applicationSchedules = null;
```

Comment:
In case of an exception we should at least log or throw the same to keep track of what went wrong.